src/status_im/contexts/settings/wallet/saved_addresses/sheets/share_address/view.cljs

```diff
@@ -0,0 +1,107 @@
+    [status-im.contexts.wallet.sheets.network-preferences.view :as network-preferences]
+    [utils.i18n :as i18n]
+    [utils.image-server :as image-server]
+    [utils.re-frame :as rf]))
+
+(def qr-size 500)
+
+(defn- navigate-back
+  []
+  (rf/dispatch [:navigate-back]))
+
+(defn- share-action
+  [address share-title]
+  (rf/dispatch [:open-share
+                {:options (if platform/ios?
+                            {:activityItemSources [{:placeholderItem {:type    :text
+                                                                      :content address}
+                                                    :item            {:default {:type :text
+                                                                                :content
+                                                                                address}}
+                                                    :linkMetadata    {:title share-title}}]}
+                            {:title     share-title
+                             :subject   share-title
+                             :message   address
+                             :isNewTask true})}]))
+
+(defn- open-preferences
+  [selected-networks set-selected-networks]
+  (let [on-save       (fn [chain-ids]
+                        (rf/dispatch [:hide-bottom-sheet])
+                        (set-selected-networks (map network-utils/id->network chain-ids)))
+        sheet-content (fn []
+                        [network-preferences/view
+                         {:blur?             true
+                          :selected-networks (set selected-networks)
+                          :on-save           on-save
+                          :button-label      (i18n/label :t/display)}])]
+    (rf/dispatch [:show-bottom-sheet
+                  {:theme   :dark
+                   :shell?  true
+                   :content sheet-content}])))
+
+(defn view
+  []
+  (let [padding-top                                (:top (safe-area/get-insets))
+        {:keys [name address customization-color]} (rf/sub [:get-screen-params])
+        [wallet-type set-wallet-type]              (rn/use-state :legacy)
+        [selected-networks set-selected-networks]  (rn/use-state constants/default-network-names)
+        on-settings-press                          (rn/use-callback #(open-preferences
+                                                                      selected-networks
+                                                                      set-selected-networks)
+                                                                    [selected-networks])
+        on-legacy-press                            (rn/use-callback #(set-wallet-type :legacy))
+        on-multichain-press                        (rn/use-callback #(set-wallet-type :multichain))
+        preferred-networks                         (rf/sub [:wallet/preferred-chain-names-for-address
+                                                            address])
```

Comment:
This shouldn't be used as this sub is used for wallet accounts, not for saved addresses.
If you need the networks, a separate key `network-preferences-names` in the saved address map will give you the network names.